File: src/BetterLyrics.DotNet/BetterLyrics.Core/ViewModels/PlayQueueViewModel.cs
using BetterLyrics.Core.Interfaces.Services;
using BetterLyrics.Core.Models.Settings;
using CommunityToolkit.Mvvm.ComponentModel;

namespace BetterLyrics.Core.ViewModels;

public partial class PlayQueueViewModel : BaseViewModel
{
    private readonly ISettingsService _settingsService;

    public PlayQueueViewModel(ISmtcService smtcService, ISettingsService settingsService)
    {
        _settingsService = settingsService;
        SMTCService = smtcService;
        AppSettings = _settingsService.AppSettings;
    }

    public ISmtcService SMTCService { get; set; }

    [ObservableProperty] public partial AppSettings AppSettings { get; set; }
}

File: src/BetterLyrics.KMP/shared/src/commonMain/kotlin/com/jayfunc/betterlyrics/viewmodels/PlayQueueViewModel.kt
package com.jayfunc.betterlyrics.viewmodels

import androidx.lifecycle.ViewModel
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.getValue
import androidx.compose.runtime.setValue
import com.jayfunc.betterlyrics.interfaces.services.ISmtcService
import com.jayfunc.betterlyrics.interfaces.services.ISettingsService
import com.jayfunc.betterlyrics.models.settings.AppSettings

class PlayQueueViewModel(
    val smtcService: ISmtcService,
    private val settingsService: ISettingsService
) : ViewModel() {
    var appSettings: AppSettings? by mutableStateOf(settingsService.appSettings)
}
